src/scripts/ui/system-status.js

```diff
@@ -3,98 +3,92 @@ window.addEventListener('DOMContentLoaded', () => {
+      title: 'Partial Service Disruption',
+    },
+    500: {
+      icons: $warningIcons,
       message: 'Service Disruption - Our web services are temporarily unavailable.',
+      title: 'Service Disruption',
+    },
+    600: {
+      icons: $warningIcons,
+      message: 'Security Event - We are currently mitigating issues relating to some of our web services.',
+      title: 'Security Event',
     },
   };
 
-  const setSystemStatus = ($icons, message) => {
+  const setSystemStatus = (status) => {
     $allSystemStatusIcons.forEach($systemStatusIcon => {
       if ($systemStatusIcon) {
         $systemStatusIcon.classList.remove('show-status-icon');
       }
     });
 
-    // This should probably be cleaned up, but the operational icon appears on the
-    // page only once, whereas the other icons show twice. All are hidden on page load
-    // using display: none;.
-
-    if ($icons.length > 1) {
+    if (status.icons.length > 1) {
       $headerStatus.classList.add('show-header-system-status');
-      $headerStatusMessage.innerText = message;
+      $headerStatusMessage.innerText = `${status.title}: ${status.message}`;
     } else {
       $headerStatus.classList.remove('show-header-system-status');
     }
 
-    $icons.forEach((icon) => {
+    status.icons.forEach((icon) => {
       icon.classList.add('show-status-icon');
     });
   };
 
   const getSystemStatus = () =>
     // eslint-disable-next-line compat/compat
     fetch('https://status.maxmind.com/1.0/status/53fcfbb2ac0c957972000235')
-      .then((res) => res.json())
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        throw new Error('status document could not be fetched');
+      })
       .then((json) => {
-        const systemStatusCode = json.result.status_overall.status_code;
-        switch (systemStatusCode) {
-          case 100:
-            setSystemStatus(status.OPERATIONAL.icons, status.OPERATIONAL.message);
-            break;
-          case 200:
-            setSystemStatus(status.PLANNED_MAINTENANCE.icons, status.PLANNED_MAINTENANCE.message);
-            break;
-          case 300:
-            setSystemStatus(status.DEGRADED_PERFORMANCE.icons, status.DEGRADED_PERFORMANCE.message);
-            break;
-          case 400:
-            setSystemStatus(status.PARTIAL_SERVICE_DISRUPTION.icons, status.PARTIAL_SERVICE_DISRUPTION.message);
-            break;
-          case 500:
-            setSystemStatus(status.SERVICE_DISRUPTION.icons, status.SERVICE_DISRUPTION.message);
-            break;
-          case 600:
-            setSystemStatus(status.SECURITY_EVENT.icons, status.SECURITY_EVENT.message);
-            break;
+        const statusCode = json.result.status_overall.status_code;
+        if (json.result.incidents.length !== 0) {
+          setSystemStatus({
+            icons: $warningIcons,
+            message: json.result.incidents[0].name,
+            title: status[Number(statusCode)].title,
```

Comment:
`Number()` here is not needed. Object keys are always strings in JS, so despite using numbers to declare the keys for `status`, they are converted to strings.